src/overlays/actors/ovl_Bg_Hidan_Curtain/z_bg_hidan_curtain.c

```diff
@@ -21,25 +46,194 @@ const ActorInit Bg_Hidan_Curtain_InitVars = {
     (ActorFunc)BgHidanCurtain_Update,
     (ActorFunc)BgHidanCurtain_Draw,
 };
-*/
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Hidan_Curtain/BgHidanCurtain_Init.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Hidan_Curtain/BgHidanCurtain_Destroy.s")
+void BgHidanCurtain_Init(Actor* thisx, GlobalContext *globalCtx) {
+    s32 pad;
+    BgHidanCurtain* this = THIS;
+    BgHidanCurtainParams* sp24;
+
+    osSyncPrintf("Curtain (arg_data 0x%04x)\n", this->actor.params);
+    Actor_SetHeight(&this->actor, 20.0f);
+    this->type = (thisx->params >> 0xC) & 0xF; // needed for matching
+    if (this->type >= 7) {
+        // Type is not set
+        osSyncPrintf("Error : object のタイプが設定されていない(%s %d)(arg_data 0x%04x)\n", "../z_bg_hidan_curtain.c", 352, this->actor.params);
+        Actor_Kill(&this->actor);
+        return;
+    }
+
+    this->isSmall = ((this->type == 2) || (this->type == 4)) ? 1 : 0;
```

Comment:
since `isSmall` is used as an index rather than as a boolean maybe naming it like `sizeIdx` makes more sense
Or rather `paramsIdx` since it indexes into `sHCParams` and the size is only the end result of the params